packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -1126,6 +1190,7 @@ export class ApexDebug extends LoggingDebugSession {
       );
       variables.sort(ApexVariable.compareVariables);
       response.body = { variables: variables };
+      this.resetIdleTimer();
```

Comment:
Reset idle timers when a variable is inspected (either by expanding a variable or pausing the apex runtime).

Reply:
This is very visible in the actual debugger logic. I think it would be better to move the timeout lifecycle handling to a lower level. For example, what about overriding `dispatchRequest`? This would reset the timeout on every request from VS Code.

How is the timeout handle on the server? Would it also make sense to reset timers in `handleEvent`, i.e. whenever we receive an event from the server?

What about `handleSessionTerminated`? Shouldn't this clear timers when the server terminates a session?

Reply:
@guw The server side inserts a heartbeat whenever it gets the state or a frame. A debugger client gets the state when we stop (because of either a breakpoint or step command). A debugger client gets the frame if the user selects a different frame to inspect. Either way, those actions involve getting variables which is why I put the reset in `variablesRequest`.

`handleSessionTerminated` sends a `TerminatedEvent` to VS Code, which in turn will send a `disconnectRequest` to the adapter which already clears the timer.